src/components/Card.js

```diff
@@ -0,0 +1,76 @@
+export default class Card {
+  //good to go 4.30, idk what the reviewer means about changing profile elements in here
+  constructor(
+    data,
+    cardSelector,
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  ) {
+    this._data = data;
+    this._name = data.name;
+    this._link = data.link;
+    this._cardSelector = cardSelector;
+    this._handleImageClick = handleImageClick;
+    this._handleDeleteCard = handleDeleteCard;
+    this._handleLike = handleLike;
+    this._id = data._id;
+    this._isLiked = data._isLiked;
+  }
+
+  _setEventListeners() {
+    this._cardElement
+      .querySelector(".card__like-button")
+      .addEventListener("click", () => {
+        this._handleLike(this);
+      });
+    this._cardElement
+      .querySelector(".card__trash-button")
+      .addEventListener("click", () => {
+        this._handleDeleteCard(this._id);
+      });
+    this._cardElement
+      .querySelector(".card__image")
+      .addEventListener("click", () => {
+        this._handleImageClick({ name: this._name, link: this._link });
+      });
+  }
+
+  handleLikeIcon() {
```

Comment:
Needs correction: please remove this extra function and create the handler for updating the likes state instead (it should update isLiked state and render the likes)

```js
// call this method from index.js after successful api response in `then` block
setIsLiked(isLiked) { 
   // set instance variable
    this._isLiked = isLiked; 
    this._renderLikes(); 
} 

// index.js
if (card.isLiked()) {
  api.dislikeCard(...).then(response => card.setIsLiked(response._isLiked)).catch(() => /*handle error*/)
} else {
 // the same for adding like
}
```

Reply:
Still needs correction